src/awx_plugins/interfaces/_temporary_private_inject_api.py

```diff
@@ -287,7 +288,10 @@ def inject_credential(
         )
         if extra_vars:
             path = _build_extra_vars_file(extra_vars, private_data_dir)
-            container_path = get_incontainer_path(path, private_data_dir)
+            container_root = kwargs.get('container_root', '')
+            container_path = get_incontainer_path(
+                path, private_data_dir, container_root=container_root,
```

Comment:
The MyPy error here (reported @ https://results.pre-commit.ci/run/github/858143066/1741025635.F2wg0bCpRjSZ-U2dUypT0w) is caused by using `kwargs.get()` since its type declaration says that it can basically return `typing.Any` which is not `str`. And so it doesn't like letting you pass arbitrarily-typed things where only strings should fit.

Specifically, it's defined in https://github.com/python/typeshed/blob/6f98c59/stdlib/builtins.pyi#L1144-L1150. If kwargs were to be a typed dict, it'd narrow the type properly.

Reply:
Marking this as resolved.